tsdb/engine/tsm1/array_cursor.gen.go.tmpl

```diff
@@ -288,9 +292,9 @@ func (c *{{$type}}) Next() {{$arrayType}} {
 		}
 	}
 
-	if pos > 0 && c.res.Timestamps[pos-1] <= c.end {
+	if pos > 0 && c.res.Timestamps[pos-1] < c.end {
 		pos -= 2
-		for pos >= 0 && c.res.Timestamps[pos] <= c.end {
+		for pos >= 0 && c.res.Timestamps[pos] < c.end {
```

Comment:
DescendingArrayCursors need to include points that fall on the left side of the range boundary.

Reply:
I have the same question here as above, this is a port of a bug fix from elsewhere, right?

Reply:
Correct.